tst/testinstall/format.tst

```diff
@@ -0,0 +1,56 @@
+gap> START_TEST("format.tst");
+gap> StringFormatted("a{}b{}c{}d", 1,(),"xyz");
+"a1b()cxyzd"
+gap> StringFormatted("{}{}{}", 1,(),"xyz");
+"1()xyz"
+gap> StringFormatted("{{}}{}}}{{", 0);
+"{}0}{"
+gap> StringFormatted("{", 1);
+Error, Invalid format string, no matching '}' at position 1
+gap> StringFormatted("{abc", 1);
+Error, Invalid format string, no matching '}' at position 1
+gap> StringFormatted("}", 1);
+Error, Mismatched '}' at position 1
+gap> StringFormatted("{3}{2}{2}{3}{1}", 1,2,3,4);
+"32231"
+gap> StringFormatted("{3}{}{2}{}{1}", 1,2,3,4);
+"31221"
+gap> StringFormatted("{a}{b}{a}", rec(a := (1,2), b := "ch"));
+"(1,2)ch(1,2)"
+gap> StringFormatted("{a}{b}{a}", 1,2);
+Error, first data argument must be a record when using {a}
+gap> StringFormatted("{}", rec());
+"rec(  )"
+gap> StringFormatted("{1}", rec());
+"rec(  )"
+gap> r1 := SymmetricGroup([3..5]);;
+gap> r2 := AlternatingGroup([1,3,5]);;
+gap> r3 := AlternatingGroup([11,12,13]);;
+gap> StringFormatted("{1!s} {1!v} {1!d}", r1);
+"SymmetricGroup( [ 3 .. 5 ] ) Sym( [ 3 .. 5 ] ) <object>\n"
+gap> StringFormatted("{!s} {!v} {!d}", r1, r2, r3);
+"SymmetricGroup( [ 3 .. 5 ] ) Alt( [ 1, 3 .. 5 ] ) <object>\n"
+gap> StringFormatted("{a!s} {b!v} {c!d}", rec(a := r1, b := r2, c := r3));
+"SymmetricGroup( [ 3 .. 5 ] ) Alt( [ 1, 3 .. 5 ] ) <object>\n"
+gap> StringFormatted("{a!x}", rec(a := r1));
```

Comment:
I find it difficult to read this test file, and to see what's there -- so if I wanted to add more tests, I'd struggle to decide where to do so.

Perhaps consider structuring this test file a bit by inserting some empty lines to separate tests into "logical" groups? (Well, empty lines followed by a comment, to suppress whitespace diffs in the test; like this:
```
gap> StringFormatted("{1}", rec());
"rec(  )"

# test format options
gap> r1 := SymmetricGroup([3..5]);;
gap> r2 := AlternatingGroup([1,3,5]);;
```